### PR TITLE
Bring changes done in 21.11 by WNPRC staff

### PR DESCRIPTION
**Pull request by dnicolalde**
#### Rationale
We are moving to use develop branch in our repo, we want to make sure we keep up to date with changes done in your develop branch. Created a develop branch from your fork and merge all the changes done to our 21.11 staging branch. We are no longer going to use the staging branch, all long term project will be develop against our develop branch. 

Quick fixes will be features branch from releaseXX.X-SNAPSHOT and merge into the original branch and port forward into develop.

#### Related Pull Requests
* 

#### Changes
*

### CODE REVIEW

File: WNPRC_EHR/resources/queries/study/wnprcFullHistory/Full History.qview.xml [hunk 1]
```diff
@@ -19,7 +19,7 @@
        <sort column="date" descending="true"/>
    </sorts>
    <filters>
-       <filter column="dataset/name" operator="in" value="assignments;behavetrem;biopsy;blood;clinremarks;deaths;encounters;drug;housing;necropsy;problem;weight;tb;clinpathRuns;foodDeprives;ultrasounds;pregnancy_outcomes;breeding_encounters,chairing,restraint"/>
+       <filter column="datasetname" operator="in" value="assignments;behavetrem;biopsy;blood;clinremarks;deaths;encounters;drug;housing;necropsy;problem;weight;tb;clinpathRuns;foodDeprives;ultrasounds;pregnancy_outcomes;breeding_encounters;research_ultrasounds,chairing,restraint;watertotal"/>
```
Comment (labkey-bpatel) [inline WNPRC_EHR/resources/queries/study/wnprcFullHistory/Full History.qview.xml:22]: Should this be just 'dataset' or 'dataset/name' or 'dataset/label' - i don't see a 'datasetname' column in the columns listed above.

File: WNPRC_EHR/src/org/labkey/wnprc_ehr/pages/query/actions.ts [hunk 2]
```diff
@@ -0,0 +1,122 @@
...
+           success: data => {
+             console.log("success");
+             resolve(data);
+           },
+           failure: data => {
+             console.log("error");
+             console.log(data);
+             reject(data);
+           }
+         });*/
```
Comment (labkey-bpatel) [inline WNPRC_EHR/src/org/labkey/wnprc_ehr/pages/query/actions.ts:28]: Commented out code, but looks like saveWeightActionSaveRowsDirect() isnt being used

File: WNPRC_Purchasing/src/client/RequestEntry/RequestEntry.tsx [hunk 3]
```diff
@@ -149,8 +149,8 @@ export const App: FC = memo(() => {
                 const lineItemVals = await getData(
                     'ehr_purchasing',
                     'lineItems',
-                    'rowId, requestRowId, item, itemUnitId, controlledSubstance, quantity, quantityReceived, unitCost, itemStatusId',
-                    undefined,
+                    'rowId, requestRowId, lineItemNumber, item, itemUnitId, controlledSubstance, quantity, quantityReceived, unitCost, itemStatusId',
+                    'lineItemNumber',
```
Comment (labkey-bpatel) [inline WNPRC_Purchasing/src/client/RequestEntry/RequestEntry.tsx:153]: lineItemNumber listed twice

File: WNPRC_EHR/resources/queries/study/wnprcFullHistory/Full History.qview.xml (hunk 1 above)
Comment (dnicolalde) [reply WNPRC_EHR/resources/queries/study/wnprcFullHistory/Full History.qview.xml:22]: We made a change to the wnprcFullHistory.sql which is here:
WNPRC_EHR/resources/queries/study/wnprcFullHistory.sql
And it should match datasetname

File: WNPRC_EHR/src/org/labkey/wnprc_ehr/pages/query/actions.ts (hunk 2 above)
Comment (dnicolalde) [reply WNPRC_EHR/src/org/labkey/wnprc_ehr/pages/query/actions.ts:28]: @csebranek Can you look into this?

File: WNPRC_Purchasing/src/client/RequestEntry/RequestEntry.tsx (hunk 3 above)
Comment (dnicolalde) [reply WNPRC_Purchasing/src/client/RequestEntry/RequestEntry.tsx:153]: We added sort to the getData function

File: WNPRC_EHR/src/org/labkey/wnprc_ehr/pages/query/actions.ts (hunk 2 above)
Comment (csebranek) [reply WNPRC_EHR/src/org/labkey/wnprc_ehr/pages/query/actions.ts:28]: correct, it is not being used and can be safely deleted

**Review by labkey-bpatel — APPROVED** [review]

**Pull request merged**